src/css.js

```diff
@@ -133,21 +130,33 @@ function getWidthOrHeight( elem, dimension, extra ) {
 		val = "auto";
 	}
 
-	// Check for style in case a browser which returns unreliable values
-	// for getComputedStyle silently falls back to the reliable elem.style
-	valueIsBorderBox = valueIsBorderBox &&
-		( support.boxSizingReliable() || val === elem.style[ dimension ] );
+	isBorderBox = jQuery.css( elem, "boxSizing", false, styles ) === "border-box";
+	valueIsBorderBox = isBorderBox && support.boxSizingReliable();
+	offsetProp = "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 );
 
 	// Fall back to offsetWidth/offsetHeight when value is "auto"
 	// This happens for inline elements with no explicit setting (gh-3571)
 	// Support: Android <=4.1 - 4.3 only
 	// Also use offsetWidth/offsetHeight for misreported inline dimensions (gh-3602)
-	if ( val === "auto" ||
-		!parseFloat( val ) && jQuery.css( elem, "display", false, styles ) === "inline" ) {
-
-		val = elem[ "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 ) ];
+	// Support: IE 9-11 only
+	// Also use offsetWidth/offsetHeight for when box sizing is unreliable
+	// Do not use offset for elements that don't have it (e.g. SVG)
+	if ( typeof elem[ offsetProp ] !== "undefined" &&
```

Comment:
Looked at this again. That's actually the same behavior as before, although it's harder to see. We apparently want it to be false there. I shouldn't have said that the computed value for SVG could be trusted to be border box in IE 9-11.

Reply:
But SVG elements _do_ respect box-sizing in IE: https://jsbin.com/zafagupixi
Also—and I had forgotten this until just now—`boxSizingUnreliable` really seems to track a specific misbehavior in the form of always reporting content box dimensions, which we can probably use to our advantage here.

But if you really want to use offsetWidth/offsetHeight, what would you say to a variation on something like this?
```js
// Fall back to offsetWidth/offsetHeight when value is "auto"
// This happens for inline elements with no explicit setting (gh-3571)
// Support: IE 9-11 only
// Also use offsetWidth/offsetHeight for when box sizing is unreliable
// Support: Android <=4.1 - 4.3 only
// Also use offsetWidth/offsetHeight for misreported inline dimensions (gh-3602)
if ( isBorderBox !== valueIsBorderBox || val === "auto" ||
		!parseFloat( val ) && jQuery.css( elem, "display", false, styles ) === "inline" ) {

	// Some elements don't have offsetWidth/offsetHeight (e.g., SVG)
	// But where available, it is a means of reading (non-fractional) border box dimensions
	if ( elem[ offsetProp ] > 0 ) {
		val = elem[ offsetProp ];
		valueIsBorderBox = true;
	}
}
```

Reply:
Are you sure you tried that in IE? I see a different result than in Chrome.

Reply:
Or maybe I'm confusing your point. Are we saying the same thing in different ways? The render seems fine, it's just that computed value for border-box is content-box for SVG. Your code suggestion should work.

Reply:
Actually, I just tried that and it broke the inline/disconnected tests. We want `valueIsBorderBox` to still be set to true for inline and disconnected elements, but false for SVG.

Reply:
I don't think a statement like "we want `valueIsBorderBox` to still be set to true for inline and disconnected elements, but false for SVG" is accurate. What we want is proper input for `boxModelAdjustment`, which returns the amount by which the width or height of a specified box (conveyed by `box`) exceeds a CSS computed width or height (subject to a box model indicated by `isBorderBox`). `boxModelAdjustment` could probably use some clarifying renames and/or comments, but the intended returned values are as follows:

`box` ＼ `isBorderBox` | false (`content-box`) | true (`border-box`)
---- | ---- | ----
content | 0 | -border - padding
padding | padding (+ scroll gutter) | -border
border | padding + border (+ scroll gutter) | 0
margin | padding + border + margin (+ scroll gutter) | margin

The [tests that failed](https://github.com/jquery/jquery/blob/6153eb0fd401cda90bf2007335cd4338093d38f0/test/unit/dimensions.js#L415) are... bad. They were only passing before this PR because of the inline style check that you're removing, and they're only passing in the current state of this PR because it includes an implicit border-box bias when _attempting_ to use offsetWidth/offsetHeight, even if that attempt fails.

We make no promises about accuracy in those disconnected/hidden cases, but if we want to continue doing our best then the right way would be to trust the computed box-sizing value for disconnected elements.

Reply:
@timmywil this should work, and at a reasonable +9 bytes over master:
```js
var styles = getStyles( elem ),
	val = curCSS( elem, dimension, styles ),
	isBorderBox = jQuery.css( elem, "boxSizing", false, styles ) === "border-box",
	valueIsBorderBox = isBorderBox,
	offsetProp = "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 );
…
// Fall back to offsetWidth/offsetHeight when value is "auto"
// This happens for inline elements with no explicit setting (gh-3571)
// Support: IE 9-11 only
// Also use offsetWidth/offsetHeight for when box-sizing is unreliable
// Support: Android <=4.1 - 4.3 only
// Also use offsetWidth/offsetHeight for misreported inline dimensions (gh-3602)
if ( isBorderBox && !support.boxSizingReliable() || val === "auto" ||
	!parseFloat( val ) && jQuery.css( elem, "display", false, styles ) === "inline" ) {

	if ( elem.getClientRects().length ) {

		// Where available, offsetWidth/offsetHeight approximate border box dimension.
		// Where not available (e.g., SVG), assume unreliable box-sizing and interpret the
		// retrieved value as a content box dimension.
		valueIsBorderBox = offsetProp in elem;
		if ( valueIsBorderBox ) {
			val = elem[ offsetProp ];
		}
	}
}
```
(note the initial assignment of `valueIsBorderBox` to match `isBorderBox`, which will always be preserved for hidden/disconnected elements but will be overridden for visible elements when box-sizing makes gCS dimensions unreliable—matching IE9-11 behavior)

Reply:
My statement is still accurate tho, right? The point was simply that the computed value can be trusted for hidden/disconnected but not for SVG, which is maintained in your suggestion. But this does avoid accessing offset for hidden/disconnected when unnecessary so I like it better.

Reply:
Updated. I just moved the rects check to the outer if statement as it was the same size.